rust/lance-io/src/ffi.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use arrow::ffi_stream::FFI_ArrowArrayStream;
+use arrow_array::RecordBatch;
+use arrow_schema::{ArrowError, SchemaRef};
+use futures::StreamExt;
+use lance_core::Result;
+
+use crate::stream::RecordBatchStream;
+
+#[pin_project::pin_project]
```

Comment:
I think you either need the `+ Unpin` bound OR `pin_project`.  I don't think you should need both.  Can you remove one or the other?